Kinvey/Kinvey/RealmCache.swift

```diff
@@ -747,7 +743,21 @@ extension RealmCache: DynamicCacheType {
     }
     
     internal func cascadeDelete(realm: Realm, entityType: String, entity: Object, deleteItself: Bool = true) {
-        if let schema = realm.schema[entityType] {
+        if let cascadeDeletable = entity as? CascadeDeletable {
+            if let entity = entity as? Entity {
+                if let acl = entity.acl {
+                    realm.delete(acl)
+                }
+                if let metadata = entity.metadata {
+                    realm.delete(metadata)
+                }
+            }
+            try! cascadeDeletable.cascadeDelete(executor: RealmCascadeDeleteExecutor(realm: realm))
```

Comment:
the `cascadeDelete()` method allows each class delete their own nested objects, so to avoid mistakes and leave dirty data (and eventually run out of space), we are deleting the `acl` and `metadata` so the user does not need to do it manually.